src/DataTypes/DataTypeEnum.cpp

```diff
@@ -191,6 +191,33 @@ static void checkASTStructure(const ASTPtr & child)
                         ErrorCodes::UNEXPECTED_AST_STRUCTURE);
 }
 
+static void autoAssignNumberForEnum(const ASTPtr & arguments)
+{
+    UInt64 literal_child_count = 0;
+    UInt64 func_child_count = 0;
+    ASTs assign_number_child;
+    assign_number_child.reserve(arguments->children.size());
+    for (const ASTPtr & child : arguments->children)
+    {
+        if (child->as<ASTLiteral>())
+        {
+            ASTPtr func = makeASTFunction("equals", child, std::make_shared<ASTLiteral>(++literal_child_count));
+            assign_number_child.emplace_back(func);
+        }
+        else
+        {
+            ++func_child_count;
+            assign_number_child.emplace_back(child);
+        }
+    }
+
+    if (func_child_count > 0 && literal_child_count > 0)
+        throw Exception("ALL Elements of Enum data type must be of form: 'name' = number or 'name', where name is string literal and number is an integer",
```

Comment:
Strange exception message, wrong syntax.

Why ALL is all caps?
Why Element with capital E?

Reply:
missed this one...

Reply:
I will fix it, I just want to emphasize it....